### PR TITLE
tests: Add InvalidBarriers layer validation test

### PR DESCRIPTION
**Pull request by Tony-LunarG**
Add a test checking for 8 ways to get INVALID_BARRIER.  Tested on 16.04 Mesa and Win 7 Nvidia

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -4831,6 +4831,144 @@ TEST_F(VkLayerTest, BufferMemoryBarrierNoBuffer) {
...
+    img_barrier.subresourceRange.layerCount = 1;
+    img_barrier.subresourceRange.levelCount = 1;
+    vkCmdPipelineBarrier(m_commandBuffer->GetBufferHandle(),
+                         VK_PIPELINE_STAGE_HOST_BIT,
+                         VK_PIPELINE_STAGE_VERTEX_SHADER_BIT, 0, 0, nullptr, 0,
+                         nullptr, 1, &img_barrier);
+    m_errorMonitor->VerifyFound();
+    img_barrier.newLayout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT,
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: please add comment for this particular case

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -4831,6 +4831,144 @@ TEST_F(VkLayerTest, BufferMemoryBarrierNoBuffer) {
...
+                                         "baseArrayLayer");
+    img_barrier.subresourceRange.baseArrayLayer = 1;
+    vkCmdPipelineBarrier(m_commandBuffer->GetBufferHandle(),
+                         VK_PIPELINE_STAGE_HOST_BIT,
+                         VK_PIPELINE_STAGE_VERTEX_SHADER_BIT, 0, 0, nullptr, 0,
+                         nullptr, 1, &img_barrier);
+    m_errorMonitor->VerifyFound();
+    img_barrier.subresourceRange.baseArrayLayer = 0;
+
+    m_errorMonitor->SetDesiredFailureMsg(
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: ditto

**Comment by tobine** [comment]
LGTM :+1:

There were two subcases within the test that weren't commented so would be nice to see comments for those cases as well.

**Comment by Tony-LunarG** [comment]
Added comments and pushed to master

**Pull request closed without merging**